alerter/src/main/java/com/tapadoo/alerter/AlerterForDialog.kt

```diff
@@ -0,0 +1,817 @@
+     */
+    fun setProgressColorRes(@ColorRes color: Int): AlerterForDialog {
+        alert?.setProgressColorRes(color)
+
+        return this
+    }
+
+    /**
+     * Set the Progress bar color from a color resource
+     *
+     * @param color The color resource
+     * @return This AlertForDialog
+     */
+    fun setProgressColorInt(@ColorInt color: Int): AlerterForDialog {
+        alert?.setProgressColorInt(color)
+
+        return this
+    }
+
+    /**
+     * Set if the Alert is dismissible or not
+     *
+     * @param dismissible true if it can be dismissed
+     * @return This AlertForDialog
+     */
+    fun setDismissable(dismissible: Boolean): AlerterForDialog {
+        alert?.setDismissible(dismissible)
+
+        return this
+    }
+
+    /**
+     * Set a Custom Enter Animation
+     *
+     * @param animation The enter animation to play
+     * @return This AlertForDialog
+     */
+    fun setEnterAnimation(@AnimRes animation: Int): AlerterForDialog {
+        alert?.enterAnimation = AnimationUtils.loadAnimation(alert?.context, animation)
+
+        return this
+    }
+
+    /**
+     * Set a Custom Exit Animation
+     *
+     * @param animation The exit animation to play
+     * @return This AlertForDialog
+     */
+    fun setExitAnimation(@AnimRes animation: Int): AlerterForDialog {
+        alert?.exitAnimation = AnimationUtils.loadAnimation(alert?.context, animation)
+
+        return this
+    }
+
+    /**
+     * Show a button with the given text, and on click listener
+     *
+     * @param text The text to display on the button
+     * @param onClick The on click listener
+     */
+    fun addButton(
+            text: CharSequence, @StyleRes style: Int = R.style.AlertButton,
+            onClick: View.OnClickListener
+    ): AlerterForDialog {
+        alert?.addButton(text, style, onClick)
+
+        return this
+    }
+
+    /**
+     * Set the Button's Typeface
+     *
+     * @param typeface Typeface to use
+     * @return This AlertForDialog
+     */
+    fun setButtonTypeface(typeface: Typeface): AlerterForDialog {
+        alert?.buttonTypeFace = typeface
+
+        return this
+    }
+
+    fun getLayoutContainer(): View? {
+        return alert?.layoutContainer
+    }
+
+    /**
+     * Creates a weak reference to the calling Dialog
+     *
+     * @param context The calling Dialog
+     */
+    private fun setActivity(context: Context) {
+        activityWeakReference = WeakReference(context)
+    }
+
+    companion object {
+
+        private var activityWeakReference: WeakReference<Context>? = null
+        private var mAppCompatDialog: AppCompatDialog? = null
+
+
+        @JvmStatic
+        fun create(context: Context?, view: AppCompatDialog): AlerterForDialog {
+            return create(context, R.layout.alerter_alert_default_layout, view)
+        }
+
+        @JvmStatic
+        fun create(context: Context?, @LayoutRes layoutId: Int, view: AppCompatDialog): AlerterForDialog {
```

Comment:
I think the right approach would be to refactor this into the original Alerter class, allowing the caller to use either and activity, or dialog to display the alert.

Reply:
changed